src/test/java/io/javalin/TestRequestTooLong.kt

```diff
@@ -0,0 +1,43 @@
+package io.javalin
+
+import io.javalin.util.TestUtil.test
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.junit.Test
+
+class TestRequestTooLong {
+
+    @Test
+    fun `throws exception when request cache is too small for request body` () {
+        val javalin = Javalin.create().enableRequestTooLongException().maxBodySizeForRequestCache(10)
+        test(javalin) { app, http ->
+            app.post("/long-post-body") {ctx ->
+                val bodyStringFirstRead = ctx.body()
+                val bodyString = ctx.body()
+                ctx.result(bodyString)
+            }
+            app.exception(Exception::class.java) { e, ctx ->
+                ctx.result(e.toString())
+            }
+
+            val longPostBody = "{\"some_long_string\":\"loooooongteext\"}";
+            assertThat(http.post("/long-post-body").body(longPostBody).asString().body, `is`("java.lang.Exception: Request cache size is too small for request body.") )
```

Comment:
checking for the exact exception message feels brittle and not very clean too me.

Reply:
fixed in 1fa7d2106cf60572c86edcef229c77e667e561df